src/MixpanelEventForwarder.js

```diff
@@ -137,11 +146,87 @@ var constructor = function () {
         }
     }
 
-    function onUserIdentified(user) {
-        var idForMixpanel;
-        var userIdentities = user.getUserIdentities()
+    function getUserIdentities(user) {
+        return user.getUserIdentities()
             ? user.getUserIdentities().userIdentities
             : {};
+    }
+
+    function onLoginComplete(user) {
+        var userIdentities = getUserIdentities(user);
+
+        // When mParticle identifies a user, the user might
+        // actually be anonymous. We only want to send an
+        // identify request to Mixpanel if the user is
+        // actually known. Only known (logged in) users
+        // will have userIdentities.
+        if (!isEmpty(userIdentities)) {
+            sendIdentifyRequest(user, userIdentities);
+        } else {
+            return 'Logged in user does not have user identities and will not be sent to Mixpanel to Identify';
+        }
+    }
+
+    function onLogoutComplete() {
+        // For all Identity Requests, we run mixpanel.identify(<user_id>)
+        // as per Mixpanel's documentation https://docs.mixpanel.com/docs/tracking-methods/identifying-users
+        // except when a user logs out, where we run mixpanel.reset() to
+        // detach the Mixpanel distinct_id from the Mixpanel device_id
+
+        if (!isInitialized) {
+            return (
+                'Cannot call logout on forwarder: ' + name + ', not initialized'
+            );
+        }
+
+        try {
+            mixpanel.mparticle.reset();
+
+            return 'Successfully called reset on forwarder: ' + name;
+        } catch (e) {
+            return 'Cannot call reset on forwarder: ' + name + ': ' + e;
+        }
+    }
+
+    function onIdentifyComplete(user) {
+        // When mParticle identifies a user, the user might
+        // actually be anonymous. We only want to send an
+        // identify request to Mixpanel if the user is
+        // actually known. Only known (logged in) users
+        // will have userIdentities.
+        var userIdentities = getUserIdentities(user);
+
+        if (!isEmpty(userIdentities)) {
+            sendIdentifyRequest(user, userIdentities);
+        } else {
+            return 'User is anonymous';
```

Comment:
I'd suggest the following to copy what you wrote for login, and also to make it easier to track back to the kit.
```suggestion
            return 'Modified user does not have user identities and will not be sent to Mixpanel to Identify';
```